File: files/400-499/407/CH-407.R
library(tidyverse)
library(readxl)

path <- "400-499/407/CH-407 Custom Grouping.xlsx"
input <- read_excel(path, range = "B3:D12")
test <- read_excel(path, range = "G3:H6")

result = input %>%
  mutate(ROW = paste0(pmin(FROM, TO), "-", pmax(FROM, TO))) %>%
  summarise(`TOTA; QUANTITY` = sum(QUANTITY, na.rm = TRUE), .by = ROW)

all.equal(result, test)
# [1] TRUE

File: files/400-499/407/CH-407.py
import pandas as pd

path = "400-499/407/CH-407 Custom Grouping.xlsx"
input = pd.read_excel(path, usecols="B:D", nrows=10, skiprows=2)
test = pd.read_excel(path, usecols="G:H", nrows=3, skiprows=2)

result = input.assign(ROW=input.apply(lambda x: f"{min(x['FROM'], x['TO'])}-{max(x['FROM'], x['TO'])}", axis=1)) \
    .groupby('ROW', as_index=False)['QUANTITY'].sum().rename(columns={'QUANTITY': 'TOTA; QUANTITY'})

print(result.equals(test))
# True